docs/index.bs

```diff
@@ -2717,12 +2725,20 @@ spec: webappsec-referrer-policy; urlPrefix: https://w3c.github.io/webappsec-refe
       : Input
       :: |serviceWorker|, a [=/service worker=]
       :: *force bypass cache for importscripts flag*, an optional flag unset by default
-      :: optional |callbackSteps|, an algorithm which takes a [=Completion=]
 
+      <div class="note">
+        Note: This algorithm typically blocks until the service worker is [=running=]. Most callsites can check for success or failure by checking if the service worker is [=running=] after this algorithm returns.
+
+        This specification generally treats an [=abrupt completion=] due to an uncaught exception the same as a normal completion. That is, if the worker throws an exception during initial script evaluation, it is still considering running and can receive events. Some callsites distinguish between these scenarios by examining the [=start status=] of the [=running=] worker.
+      </div>
+
+      1. If |serviceWorker| is [=running=], then return.
+      1. If |serviceWorker|'s [=state=] is *redundant*, then return.
+      1. Assert: |serviceWorker|'s [=start status=] is null.
       1. Let |script| be |serviceWorker|'s [=service worker/script resource=].
       1. Assert: |script| is not null.
-      1. If |serviceWorker| is already running, this algorithm must have been invoked previously. If |callbackSteps| is provided, run them with the same value as the previous time, and abort these steps.
-      1. Create a separate parallel execution environment (i.e. a separate thread or process or equivalent construct), and run the following substeps in that context:
+      1. Let |startFailed| be false.
+      1. Create a separate parallel execution environment (i.e. a separate thread or process or equivalent construct), and run the following substeps in that context [=in parallel=]:
```

Comment:
I'm not sure we need [=in parallel=] here. The steps are already running in "that context" which is a "separate parallel execution environment". Otherwise it could read like you're running steps in that context, then in parallel, which would take you off the JavaScript thread for that context.

Reply:
May as well replace "substeps" with "steps".

Reply:
Agreed, it's redundant, I remember thinking about this and somehow decided to put it in. I'll remove.